services/web/server/src/simcore_service_webserver/login/_registration_handlers.py

```diff
@@ -59,8 +62,17 @@ def _get_ipinfo(request: web.Request) -> dict[str, Any]:
 @global_rate_limit_route(number_of_requests=30, interval_seconds=MINUTE)
 async def request_product_account(request: web.Request):
     product = get_current_product(request)
+    session = await get_session(request)
+
     body = await parse_request_body_as(AccountRequestInfo, request)
     assert body.form  # nosec
+    assert body.captcha  # nosec
+
+    if body.captcha != session.get(CAPTCHA_SESSION_KEY):
```

Comment:
Why don't you directly `pop`? Note that `pop` is nothing more than `get+delete`
```python
if body.catch != session.pop(CAPTACH_SESSION_KEY, None):
    ...
```

aha... i have just noticed that you want to allow the user to retry **on the same** challenge? 
OK, in that case, just add a note to stop people like me from refactoring this and introducing a bug :-)
Something like " Challenge deleted AFTER use so it can be retried multiple times".

Consider also the possibility to create a maximum number of trials. You can control that by **adding a countdown** in `session[CAPTCHA_SESSION_KEY]` together with the text

Reply:
discussed